synapse/config/saml2_config.py

```diff
@@ -271,6 +271,24 @@ def generate_config_section(self, config_dir_path, server_name, **kwargs):
             #description: ["My awesome SP", "en"]
             #name: ["Test SP", "en"]
 
+            #ui_info:
+            #  display_name:
+            #    - lang: en
+            #      text: "Display Name is the descriptive name of your service."
+            #  description:
+            #    - lang: en
+            #      text: "Description should be a short paragraph explaining the purpose of the service."
+            #  information_url:
+            #    - lang: en
+            #      text: "https://example.com/terms-of-service"
+            #  privacy_statement_url:
+            #    - lang: en
+            #      text: "https://example.com/privacy-policy"
+            #  logo:
+            #    text: "https://example.com/logo.svg"
+            #    width: "200"
+            #    height: "80"
```

Comment:
According to https://github.com/IdentityPython/pysaml2/blob/59d6fa5df06989525d2d7e5b8762bbfa3485ab42/src/saml2/metadata.py#L268-L276 it looks as though `logo` can also be a list of dicts, which looks more flexible. Any reason that's not show in the example?

Reply:
@anoadragon453 no reasons, feel free to modify it. This is basically the config I use in my SAML config.

Reply:
@anoadragon453 like for all elements multiple languages are possible like

 ```
ui_info:
      display_name:
        - lang: en
          text: "Display Name is the descriptive name of your service."
       - lang: de
          text: "Display Name ist der beschreibende Name Ihres Dienstes"
```

So I think for `logo` it has to look like this
``` 
           logo:
               - lang: en
                 text: "https://example.com/logo_en.svg"
                 width: "200"
                 height: "80"
               - lang: de
                 text: "https://example.com/logo_de.svg"
                 width: "200"
                 height: "80"
```

Reply:
@anoadragon453 Andrew, I checked list of dicts for logo element with my IdP an this generates valid metadata `/_matrix/saml2/metadata.xml`:

saml2_config:
```
ui_info:
           [...]
           logo:
              - lang: en
                text: "https://example.com/logo_en.svg"
                width: "200"
                height: "80"
              - lang: de
                text: "https://example.com/logo_de.svg"
                width: "200"
                height: "80"
```

/saml2/metadata.xml`:
```
<ns1:Logo height="86" width="200" xml:lang="de">
https://example.com/logo_de.svg
</ns1:Logo>
<ns1:Logo height="86" width="200" xml:lang="en">
https://example.com/logo_en.svg
</ns1:Logo>
```
The central Shibboleth IdP at my university also evaluates the language for logos. If I switch the browser between DE and EN, the corresponding logo is also displayed on the registration form of the IdP.

Furthermore I have added the `keywords` element.

Reply:
Glad to hear, thank you for testing!